packages/schema/src/Schema.ts

```diff
@@ -3641,6 +3641,39 @@ export const either = <IE, E, IA, A>(
     })
   )
 
+/**
+ * @example
+ * import * as Schema from "@effect/schema/Schema"
+ *
+ * // Schema<"A" | "E", Either<"E", "A">>
+ * Schema.eitherFromUnion({
+ *   right: Schema.literal("A"),
+ *   left: Schema.literal("E")
+ * })
+ *
+ * @category Either transformations
+ * @since 1.0.0
+ */
+export const eitherFromUnion = <EI, EA, AI, AA>(schemas: {
```

Comment:
I chose to use a struct parameter  `eitherFromUnion({ left, right })` because in my use cases the schemas tend to be long and it's easier to read this way.
But it doesn't conform to the existing functions, that take two parameters `either(left, right)`.
So you may choose what you prefer and I can adjust.

Reply:
@matheuspuel let's use the standard `eitherFromUnion(left, right)` for now